File: C++/ABC/ABC006/D_10.cpp
// bit全探索 O(N・2^N)

#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++) cin >> a[i];
    int ans = n;
    if (n <= 16) {
        for (int i = 0; i < (1 << n); i++) {
            int c = 0;
            vector<int> b;
            for (int j = 0; j < n; j++) {
                if (i & (1 << j)) {
                    b.push_back(a[j]);
                    c++;
                }
            }
            bool ok = true;
            for (int i = 0; i < c - 1; i++) {
                if (b[i] > b[i + 1]) ok = false;
            }
            if (ok) {
                ans = min(ans, n - c);
            }
        }
        cout << ans << '\n';
    }
}

File: ABC/ABC006/D/D_10.py
n = int(input())
a = [int(input()) for _ in range(n)]
ans = n
for i in range(1 << n):
    c = 0
    b = []
    for j in range(n):
        if i & (1 << j):
            b.append(a[j])
            c += 1
    ok = True
    for i in range(c - 1):
        if b[i] > b[i + 1]:
            ok = False
    if ok:
        ans = min(ans, n - c)
print(ans)